src/CarbonAware.Tools/CarbonAware.Tools.WattTimeClient/src/WattTimeClient.cs

```diff
@@ -102,15 +102,15 @@ public Task<Forecast> GetCurrentForecastAsync(BalancingAuthority balancingAuthor
     }
 
     /// <inheritdoc/>
-    public async Task<IEnumerable<Forecast>> GetForecastByDateAsync(string balancingAuthorityAbbreviation, DateTimeOffset startTime, DateTimeOffset endTime)
+    public async Task<Forecast?> GetForecastOnDateAsync(string balancingAuthorityAbbreviation, DateTimeOffset generatedAt)
```

Comment:
Because we are getting a forecast on a specific date (given the `generatedAt` field), we changed the return type to be a single Forecast instead of a List. We also made it nullable because it may be the case that the user requests a forecast at a time when none was generated. This is a user input error and not a watt time error so we allow null to be a valid return type and then handle throwing the error at the data source level